compiler-rt/lib/asan/asan_fake_stack.cpp

```diff
@@ -217,26 +217,44 @@ static THREADLOCAL FakeStack *fake_stack_tls;
 FakeStack *GetTLSFakeStack() {
   return fake_stack_tls;
 }
-void SetTLSFakeStack(FakeStack *fs) {
+void SetTLSFakeStack(AsanThread* t, FakeStack* fs) {
```

Comment:
Will #163481 help to simplify this patch?

Reply:
Slightly. By only calling `SetTLSFakeStack` from `GetFakeStack` #163481 also indirectly ensures the tls var won't be set if fake stack is disabled. I added the thread arg and put the logic into `SetTLSFakeStack` to make it more explicit.